### DISCUSSION TITLE
Custom polyfill only for legacy browsers?

### DISCUSSION THREAD

**Discussion by aralroca** [vercel/next.js#11545] [Help]
I'm using custom polyfills as intersection-observer, and I added them in my project following this example:

https://github.com/zeit/next.js/blob/canary/examples/with-polyfills/next.config.js

However, these polyfills are added also in modern browsers. Is there another way to do it only for legacy browsers? (This example is very old and the things changed so fast in Next.js)

**Comment by aralroca** [comment]
Does anybody know? @Timer @timneutkens ?

**Reply by Timer** [reply]
Next.js does not currently expose a way to polyfill for legacy browsers only. This feature is on the roadmap.

**Reply by aralroca** [reply]
Ok, thank you @Timer !

**Reply by timneutkens** [reply]
Note that intersection-observer needs to be polyfilled in browsers that fall outside the nomodule window, meaning it has to be shipped to all users.
